### PR TITLE
Chore: Make detect breaking changes workflow backport compatible 

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This [PR](https://github.com/grafana/grafana/pull/44996#issuecomment-1031573343) has highlighted that always comparing against main for breaking changes won't work for backports as the PR target branch will be different.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/workflows/detect-breaking-changes-build.yml [hunk 1]
```diff
@@ -43,18 +43,18 @@ jobs:
...
-        working-directory: './main'
+        working-directory: './base'
 
     steps:
     - uses: actions/checkout@v2
       with: 
-        path: './main'
-        ref: 'main'
+        path: './base'
+        ref: ${{ github.event.pull_request.base.ref }}
```
Comment (marefr) [inline .github/workflows/detect-breaking-changes-build.yml:57]: Is there any security concern around this?

**Review by marefr — COMMENTED** [review]
Wonder if detect breaking changes should be run for anything other than main/backports. With it we could potentially catch breaking change in backport that wasn't catched comparing against main?

Added a comment

File: .github/workflows/detect-breaking-changes-build.yml (hunk 1 above)
Comment (marefr) [reply .github/workflows/detect-breaking-changes-build.yml:57]: Wasn't sure if we expose any secrets in this action. If not, should be fine

**Review by dprokop — APPROVED** [review]
lgtm!

File: .github/workflows/detect-breaking-changes-build.yml (hunk 1 above)
Comment (leventebalogh) [reply .github/workflows/detect-breaking-changes-build.yml:57]: Luckily this action uses the `pull_request` trigger which shouldn't have access to the secrets, so I think we should be safe, too. 👍

**Review by leventebalogh — APPROVED** [review]
LGTM, amazing fix 👏

**Pull request merged**